ably/realtime_client.go

```diff
@@ -73,8 +73,11 @@ func (c *Realtime) onReconnectMsg(msg *proto.ProtocolMessage) {
 				switch ch.State() {
 				case StateConnSuspended:
 					ch.attach(false)
-				case StateChanAttaching, StateChanAttached:
+				case StateChanAttaching, //(RTN19b)
+					StateChanAttached:
 					ch.mayAttach(false, false)
+				case StateChanDetached: //(RTN19b)
+					ch.mayDetach(false, false)
```

Comment:
RTN19 says "when a transport is upgraded or disconnected for any reason", but here we're reconnecting, not disconnecting. ("Upgraded" isn't really well-defined, so I'm just ignoring it for now; I've opened a Slack conversation about this.)

Reply:
I think its proper to do it here because  se resend these on new connection. We reconnect after a disconnect meaning we have a new connection by this time.

Reply:
You're right, I wasn't making much sense. We need a connection to send protocol messages!